slp_visio/slp_visio/load/strategies/connector/impl/create_connector_by_line_coordinates.py

```diff
@@ -0,0 +1,48 @@
+from typing import Optional
+
+from shapely.geometry import Point
+from vsdx import Shape
+
+from slp_visio.slp_visio.load.objects.diagram_objects import DiagramConnector
+from slp_visio.slp_visio.load.representation.simple_component_representer import SimpleComponentRepresenter
+from slp_visio.slp_visio.load.strategies.connector.create_connector_strategy import CreateConnectorStrategy
+
+
+class CreateConnectorByLineCoordinates(CreateConnectorStrategy):
+    """
+    Strategy to create a connector from the shape begin and end coordinates
+
+    We search a component shape that touch the beginning line coordinates of the given shape connector
+    We search a component shape that touch the ending line coordinates of the given shape connector
+    If we find both shapes at the end and at the beginning of the line, we create the connector
+    """
+
+    def __init__(self):
+        self.tolerance = 0.09
+        self.representer: SimpleComponentRepresenter() = SimpleComponentRepresenter()
+
+    def create_connector(self, shape: Shape, **kwargs) -> Optional[DiagramConnector]:
```

Comment:
This is not a strong opinion because I see the point of allowing different kinds of parameters in each strategy. I also agree on the connector shape is always mandatory. But, IMO, the interface is more difficult to understand and, currently, we only want to send optionally the components. My proposal is to follow the three times rule and keep the interface simpler with the signature shape: `def create_connector(self, shape: Shape, components: List[Shape] = None) -> Optional[DiagramConnector]:`.

Reply:
I agree with you that the interface is not easy to understand without taking a look at the implementations, but thinking in the next strategies in which we will need a new different parameter, the signature of the interface can grow a little bit, and each parameter will be used just in 1 strategy, e.g:
`def create_connector(self, shape: Shape, components: List[Shape] = None, trustzones: List[Shape] = None, pages: List[Page] = None, parent_shapes: List[Shape] = None, child_shapes: List[Shape] = None,  sibling_shapes: Shape) -> Optional[DiagramConnector]:`
The current approach allows us taking advantage of the Python flexibility with **kwargs, to avoid modifying the interface on each new implementation.
Let's talk about that to find the better approach.

Reply:
Finally, I've left the signature with the components parameter